### PR TITLE
LTC refactor sub.mk

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/lib/libtomcrypt/sub.mk [hunk 1]
```diff
@@ -145,11 +141,277 @@ srcs-$(_CFG_CORE_LTC_SM2_DSA) += sm2-dsa.c
...
+srcs-$(_CFG_CORE_LTC_ECC) += src/pk/ecc/ltc_ecc_mulmod.c
+srcs-$(_CFG_CORE_LTC_ECC) += src/pk/ecc/ltc_ecc_mulmod_timing.c
+srcs-$(_CFG_CORE_LTC_ECC) += src/pk/ecc/ltc_ecc_mul2add.c
+srcs-$(_CFG_CORE_LTC_ECC) += src/pk/ecc/ltc_ecc_points.c
+srcs-$(_CFG_CORE_LTC_ECC) += src/pk/ecc/ltc_ecc_projective_add_point.c
+srcs-$(_CFG_CORE_LTC_ECC) += src/pk/ecc/ltc_ecc_projective_dbl_point.c
+
+srcs-$(_CFG_CORE_LTC_ED25519) += src/pk/ec25519/ec25519_crypto_ctx.c
+srcs-$(_CFG_CORE_LTC_ED25519) += src/pk/ec25519/ec25519_export.c
+srcs-$(_CFG_CORE_LTC_ED25519) += src/pk/ec25519/tweetnacl.c
```
Comment (etienne-lms) [inline core/lib/libtomcrypt/sub.mk:350]: These 3 were built when at least one of `_CFG_CORE_LTC_X25519` and `_CFG_CORE_LTC_ED25519` was enabled.

**Review by etienne-lms — COMMENTED** [review]
Comment for commit "core: ltc: merge sub.mk's into a single sub.mk".

I see that some build directives are removed (`-Wno-unused-variable`, `-Wno-unused-parameter`). As for my review comment on `Wno-declaration-after-statement`, I'm fine with the removal but i think it should be stated in the commit message.

This change discards use of `_CFG_CORE_LTC_CIPHER`, `_CFG_CORE_LTC_AUTHENC` and `_CFG_CORE_LTC_MAC`. I suggest to also remove them from _core/crypto.mk_.

**Review by etienne-lms — COMMENTED** [review]
For commit "core: ltc: sub.mk: reorganize and simplify": 
Looks all good to me. Could apply my R-b tag but i'd rather wait for your feedback on my review comments on the 1st commit and maybe have a second look.

File: core/lib/libtomcrypt/sub.mk [hunk 2]
```diff
@@ -145,11 +141,277 @@ srcs-$(_CFG_CORE_LTC_SM2_DSA) += sm2-dsa.c
...
 srcs-$(_CFG_CORE_LTC_ED25519) += ed25519.c
 ifeq ($(_CFG_CORE_LTC_ACIPHER),y)
 srcs-y += mpi_desc.c
+cflags-mpi_desc.c-y += -Wno-declaration-after-statement
 cppflags-mpi_desc.c-y += -DMBEDTLS_ALLOW_PRIVATE_ACCESS
 endif
 
 srcs-y += tomcrypt.c
 
+srcs-$(_CFG_CORE_LTC_AES_DESC) += src/ciphers/aes/aes.c
```
[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply core/lib/libtomcrypt/sub.mk:57]: I'll fix up "core: ltc: merge sub.mk's into a single sub.mk" to keep this, and then clean it up in "core: ltc: sub.mk: reorganize and simplify"

File: core/lib/libtomcrypt/sub.mk (hunk 1 above)
Comment (jenswi-linaro) [reply core/lib/libtomcrypt/sub.mk:350]: You're right, I'll fixup the commit to use _CFG_CORE_LTC_EC25519 for this.

**Comment by jenswi-linaro** [comment]
Update. I had to do it as a force push.

**Comment by jenswi-linaro** [comment]
I have prepared an update of LTC that I'll create a PR for once this is merged

**Comment by jenswi-linaro** [comment]
Tag applied

**Comment by jenswi-linaro** [comment]
@etienne-lms do you have any more comments?

**Review by etienne-lms — APPROVED** [review]
All good to me.
`Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>`

**Comment by jenswi-linaro** [comment]
Tag applied.

**Pull request merged**